src/newmagic.cpp

```diff
@@ -1700,6 +1708,205 @@ void cast_manipulation_spell(struct char_data *ch, int spell, int force, char *a
+        act("$n's body reels from the blast of steam.", TRUE, vict, 0, 0, TO_ROOM);
+        send_to_char("You feel lightheaded from the blast of steam.\r\n", vict);
+      } else if (GET_MENTAL(vict) - (dam * 100) <= 0) {
+        act("$n chokes and collapses to the ground, unconscious, from the steam vapours.", TRUE, vict, 0, 0, TO_ROOM);
+        send_to_char("It slams into you and robs the breath from your lungs, choking you into unconsciousness.\r\n", vict);
+      } else if (GET_MENTAL(vict) - (dam * 100) <= 300) {
+        act("$n flounders backwards, almost losing $s footing, as $e is choked by the steam cloud.", TRUE, vict, 0, 0, TO_ROOM);
+        send_to_char("You flounder backwards, feeling lightheaded as the steam threatens to choke you.\r\n", vict);
+      } else if (GET_MENTAL(vict) - (dam * 100) <= 700) {
+        act("$n steps back and coughs a few times to catch their breath.", TRUE, vict, 0, 0, TO_ROOM);
+        send_to_char("You step back as the steam hits you, feeling a little short on oxygen.\r\n", vict);
+      } else if (dam > 0) {
+        act("$n blanches as they're exposed briefly to the steam.", TRUE, vict, 0, 0, TO_ROOM);
+        send_to_char("You blanch as the feeling of hot steam washes over your face.\r\n", vict);
+      } else {
+        send_to_char("It rushes past you, not causing any damage.\r\n", vict);
+        act("$n doesn't seem hurt by your steam spell.", FALSE, vict, 0, ch, TO_VICT);
+      }
+      damage(ch, vict, dam, TYPE_MANIPULATION_SPELL, MENTAL);
+      if (IS_NPC(vict) && !IS_NPC(ch))
+        GET_LASTHIT(vict) = GET_IDNUM(ch);
+    }
+    spell_drain(ch, spell, force, basedamage);
+    break;
+  case SPELL_THUNDERBOLT:
+    if (!check_spell_victim(ch, vict, spell, arg))
+      return;
+    check_killer(ch, vict);
+    if (!AWAKE(vict))
+      target -= 2;
+    else {
+      // Dodge test: You must be awake.
+      success -= success_test(GET_DEFENSE(vict) + GET_DEFENSE(vict) ? GET_POWER(vict, ADEPT_SIDESTEP) : 0, 4 + damage_modifier(vict, buf, sizeof(buf)));
+    }
+    success += success_test(skill, 4 + target);
+      
+    send_to_char("The air around you explodes into a deafening roar!\r\n", vict);
+      
+    if (!IS_NPC(ch) && PLR_FLAGGED(ch, PLR_KILLER) && !IS_NPC(vict))
+      success = -1;
+    
+    if (success <= 0) {
+      act("Your thunderbolt spell harmlessly echoes as $n dodges it.", FALSE, vict, 0, ch, TO_VICT);
+      send_to_char("You easily dodge it!\r\n", vict);
+      act("$n dodges out of the way of the concussive blast.", TRUE, vict, 0, ch, TO_NOTVICT);
+    } else {
+      success -= success_test(GET_BOD(vict) + GET_BODY(vict), force - GET_IMPACT(vict));
+      int dam = convert_damage(stage(success, basedamage));
+      if (!AWAKE(vict)) {
+        act("$n's shudders from the concussive blast.", TRUE, vict, 0, 0, TO_ROOM);
+        send_to_char("You feel dizzy from the concussive blast.\r\n", vict);
+      } else if (GET_MENTAL(vict) - (dam * 100) <= 0) {
```

Comment:
Actually not sure what the damtype of this spell should be, arguments could be made for either one.

Reply:
Spell does Mental damage as per canon.